flake.nix

```diff
@@ -0,0 +1,144 @@
+{
+  description = "Shadysim";
+
+  inputs = {
+
+    nixpkgs.url = "github:nixos/nixpkgs/nixos-21.05";
+
+    pycrypto-src = {
+      url = "github:pycrypto/pycrypto";
+      flake = false;
+    };
+
+    simtools-src = {
+      url = "git+https://git.osmocom.org/sim/sim-tools";
+      flake = false;
+    };
+
+    imsi-pseudo-src = {
+      url = "git+https://git.osmocom.org/imsi-pseudo";
+      flake = false;
+    };
+
+  };
+
+
+  outputs =
+    { self
+    , nixpkgs
+    , pycrypto-src
+    , simtools-src
+    , imsi-pseudo-src
+    , ...
+    }:
+    let
+      supportedSystems = [ "x86_64-linux" "x86_64-darwin" ];
+      forAllSystems = f: nixpkgs.lib.genAttrs supportedSystems (system: f system);
+
+      nixpkgsFor = forAllSystems (system:
+        import nixpkgs {
+          inherit system;
+          overlays = [ self.overlay ];
+        }
+      );
+
+    in
+    {
+
+      overlay = final: prev: rec {
+
+        pycrypto =
+          with final.python2Packages;
+          buildPythonPackage {
+            pname = "pycrypto";
+            version = "2.6.1";
+            src = pycrypto-src;
+          };
+
+        shadysim =
+          with final.python2Packages;
+          buildPythonApplication {
+            pname = "shadysim";
+            version = "0.1.0";
+            src = "${simtools-src}/shadysim";
+
+            postPatch = ''
+              cp ${./files/setup.py} ./setup.py
+            '';
+
+            postInstall = ''
+              mv $out/bin/shadysim{.py,}
+            '';
+
+            propagatedBuildInputs = [
+              pycrypto
+            ];
+          };
+
+        converter =
+          let
+            CLASSPATH = "${simtools-src}/javacard/bin/converter.jar";
+            JFLAGS = "-classpath ${CLASSPATH}";
```

Comment:
why not group the two variables above into one JFLAGS ?
Also since they are not environment variables, how about not capitalizing ?

Reply:
>why not group the two variables above into one JFLAGS ?

Ah yeah, `CLASSPATH` is not required anymore. 

>Also since they are not environment variables, how about not capitalizing ?

Yep, will do.